source/nodejs/adaptivecards-designer/src/designer-peers.ts

```diff
@@ -2255,7 +2276,10 @@ export class ColumnSetPeer extends TypedCardElementPeer<Adaptive.ColumnSet> {
 
                         this.cardElement.addColumn(column);
 
-                        this.insertChild(CardDesignerSurface.cardElementPeerRegistry.createPeerInstance(this.designerSurface, this, column));
+                        let newPeer = CardDesignerSurface.cardElementPeerRegistry.createPeerInstance(this.designerSurface, this, column);
+                        newPeer.insertAfterNeighbor = true;
+
+                        this.insertChild(newPeer);
```

Comment:
this pattern feels like the `insertAfterNeighbor` state should be passed into `createPeerInstance` or `insertChild`... 

looking a little closer, is this particular to some collection-type elements? all collection-type elements? perhaps we should be applying it based on `DesignerPeerCategory` or a parameter added to `DesignerPeerRegistry.registerPeer()`?

https://github.com/microsoft/AdaptiveCards/blob/7e2306219d368a1e2ec3b641c917b9d0c4a34ef3/source/nodejs/adaptivecards-designer/src/card-designer-surface.ts#L105-L113

Reply:
I added `insertAfterNeighbor` so that we could handle adding new elements from an element action. So it is only applicable for `ImageSet`, `ColumnSet`, and `Table`

Reply:
The `onPeerAdded` callback is overloaded and is called whenever an element is being dragged over the card (**before** it is dropped) and when we add a new element via element action

https://github.com/anna-dingler/AdaptiveCards/blob/accessibilityUpdates/source/nodejs/adaptivecards-designer/src/card-designer-surface.ts#L409-L418

Reply:
@paulcam206 I might need more context here, but I don't think we can assume that we should determine `insertAfterNeighbor` based on peer type. When we drag an element over the card, `onPeerAdded` is called several times, and we don't want to find the neighbor each time.